db/User.js

```diff
@@ -1,106 +1,125 @@
-const conn = require('./conn');
+const conn = require("./conn");
 const { Sequelize } = conn;
+const { STRING, BOOLEAN } = Sequelize;
 
-const jwt = require('jsonwebtoken');
-const bcrypt = require('bcrypt');
+const jwt = require("jsonwebtoken");
+const bcrypt = require("bcrypt");
 
-
-const User = conn.define('user', {
+const User = conn.define("user", {
   username: {
-    type: Sequelize.STRING
+    type: STRING,
+    allowNull: false,
   },
   password: {
-    type: Sequelize.STRING
-  }
+    type: STRING,
+    allowNull: false,
+  },
+  isAdmin: {
+    type: BOOLEAN,
+    defaultValue: false,
+  },
+  email: {
+    type: STRING,
+    validate: {
+      isEmail: true,
+    },
+  },
+  imageUrl: {
+    type: STRING,
+    // validate:{
+    //   isUrl: true,
+    // }
+  },
+  address: {
+    type: STRING,
+  },
 });
 
-
-User.addHook('beforeSave', async(user)=> {
+User.addHook("beforeSave", async (user) => {
   user.password = await bcrypt.hash(user.password, 5);
 });
 
-User.prototype.createOrderFromCart = async function(){
+User.prototype.createOrderFromCart = async function () {
   const cart = await this.getCart();
   cart.isCart = false;
   return cart.save();
-}
+};
 
-User.prototype.addToCart = async function({ product, quantity}){
+User.prototype.addToCart = async function ({ product, quantity }) {
   const cart = await this.getCart();
   let lineItem = await conn.models.lineItem.findOne({
     where: {
       productId: product.id,
-      orderId: cart.id
-    }
+      orderId: cart.id,
+    },
```

Comment:
👍 by the way, thanks for the last element comma style.
I was trying to enable this with prettier. Will get this auto-prettier formatted in one of my earliest PRs as possible